### PR TITLE
SonarQube is add to do the analysis of Fineract code

### PR DESCRIPTION
**Pull request by IOhacker**
## Description
I am adding the support to SonarQube, there are settings that must be enabled in Travis and SonarQube.io for get it working. I have added the steps here:

https://docs.google.com/document/d/1kAoeJHxctVpNC_2b4J7o-jVALvCisf5U1yvi80Cq6uA/edit?usp=sharing

You can check the Dashboard generated:
https://sonarcloud.io/dashboard?id=IOhacker_fineract

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Commit message starts with the issue number from https://issues.apache.org/jira/projects/FINERACT/. Ex: FINERACT-646 Pockets API.

- [ ] Coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions have been followed.

- [ ] API documentation at https://github.com/apache/fineract/blob/develop/api-docs/apiLive.htm has been updated with details of any API changes.

- [ ] Integration tests have been created/updated for verifying the changes made.

- [ ] All Integrations tests are passing with the new commits.

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the list.)

Our guidelines for code reviews is at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide

### CODE REVIEW

File: fineract-provider/build.gradle [hunk 1]
```diff
@@ -145,6 +147,7 @@ group = 'org.apache.fineract'
 buildDir = new File(rootProject.projectDir, "../build")
 repositories {
 	jcenter()
+    mavenCentral()
```
Comment (vorburger) [inline fineract-provider/build.gradle:150]: Are you sure this is needed? jcentral is (should be..) a superset of mavenCentral, and already have anything that's on mavenCentral.

File: .travis.yml [hunk 2]
```diff
@@ -61,5 +61,5 @@ script:
 # using "&&" instead of several "-" means that integrationTest does not run if test fails,
 # and Docker test does not run if integration test fails, which makes PR failure easier to understand.
 # @see https://docs.travis-ci.com/user/job-lifecycle/#customizing-the-build-phase
-  - ./gradlew --console=plain licenseMain licenseTest licenseIntegrationTest check  &&  ./gradlew --console=plain integrationTest  &&  sudo service mysql stop  &&  docker-compose build  &&  docker-compose up -d  &&  sleep 30s  &&  http --verify=no --timeout 240 --check-status get https://localhost:8443/fineract-provider/actuator/health
+  - ./gradlew --console=plain licenseMain licenseTest licenseIntegrationTest check  && ./gradlew --console=plain integrationTest && ./gradlew --console=plain sonarqube -Dsonar.host.url=$SONAR_HOST_URL -Dsonar.organization=$SONAR_ORGANIZATION -Dsonar.projectKey=$SONAR_PROJECT_KEY -Dsonar.login=$SONAR_TOKEN -Dsonar.verbose=true --stacktrace &&  sudo service mysql stop  &&  docker-compose build  &&  docker-compose up -d  &&  sleep 30s  &&  http --verify=no --timeout 240 --check-status get https://localhost:8443/fineract-provider/actuator/health
```
Comment (vorburger) [inline .travis.yml]: Where/how is SONAR_TOKEN etc. defined?

**Review by vorburger — COMMENTED** [review]
Could you rebase and squash into a single commit?

I've also provided some first review feedback.

**Comment by IOhacker** [comment]
I want to add, "repositories" are duplicated I just have put jcenter and 
mavencentral inside the same {} , also the java editor show the warning, 
by the way and many others, seems that for every task inside de 
build.gradle there are duplicates .

You can find in the attached document (shared as a link) in the PR how 
to add the variables in Travis, also it goes again


https://docs.google.com/document/d/1kAoeJHxctVpNC_2b4J7o-jVALvCisf5U1yvi80Cq6uA/edit?usp=sharing

**Comment by awasum** [comment]
Please, rebase and squash these commits @itzamnamx . Also fix the conflict resulting from another merge now affecting this PR. Thanks.

**Comment by vorburger** [comment]
This was never merged, but #1893 contributed Sonar later.

**Pull request closed without merging**